### PR TITLE
feat: add bigquery cloudsql connection module

### PR DESCRIPTION
**Pull request by JoseFMP**
Please confirm that you have done the following before requesting reviews:

- [x] I have confirmed that the PR type is appropriate for the change I am making according to the [Honest Pull Request and Commit Message Naming Conventions](https://www.notion.so/honestbank/Pull-Request-and-Commit-Message-Naming-Conventions-bd97f2cbb34c4c73b1ff3a3e384b850c).
- [x] I have typed an adequate description that explains **why** I am making this change.
- [x] I have installed and run standard pre-commit hooks that lints and validates my code.

### Description

* Adds a new Terraform component module named `gcp_bigquery_cloudsql_connection`. 
* This new component is a wrapper around the resource `google_bigquery_connection` available in provider `hashicorp/google-beta`.
* This component is to be used to establish a connection on top of our Cloud SQL databases in GCP and provide connectivity into BigQuery.
* An example using the component module is added.
* Timeout of the tests is increased as this test takes longer than the other ones as it requires to build up databases.
* Linear ticket: https://linear.app/honestbank/issue/AX-52/cloudsql-bigquery-federated-query-deployment-pipeline

---
This change is [Reviewable](https://reviewable.io/reviews/honestbank/terraform-gcp-bigquery/5)

### CODE REVIEW

**Review by nithipanhn — APPROVED** [review]

**Review by lucheeseng827 — APPROVED** [review]

**Comment by MXfive** [comment]
@JoseFMP Draft is used to indicate that a PR is not yet ready for review, if this is ready then please update the status 🙏🏻

**Comment by JoseFMP** [comment]
> @JoseFMP Draft is used to indicate that a PR is not yet ready for review, if this is ready then please update the status 🙏🏻
> 
> Also if ready for review will need proper description of the changes and the mandatory linear ticket(s) are linked.

As stated in the PR status => Draft.

**Comment by MXfive** [comment]
So existing review should still be dismissed?

**Comment by JoseFMP** [comment]
> So existing review should still be dismissed? The title shouldn't be necessary, but people should not be approving PRs before they are finished.

Correct, the previously existing approvals should be dismissed as the PR was not ready.
The same happens when we add commits to existing approved PR branches, the reviews should be dismissed but is not happening.

**Review by nithipanhn — APPROVED** [review]

**Review by lucheeseng827 — APPROVED** [review]

**Review by lucheeseng827 — APPROVED** [review]

File: example/create_cloudsql_connection/main.tf [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+// create a google service account to become dataset owner
+resource "google_service_account" "owner" {
+  account_id   = "datasetowner${random_id.random_id.hex}"
+  display_name = "dataset_owner"
+}
+
+// creating the BigQuery dataset at Jakarta
+module "create_bigquery_dataset" {
```
Comment (MXfive) [inline example/create_cloudsql_connection/main.tf:12]: Should drop the `create_` prefixes from resource and module names.

Terraform code is declarative, resources and modules refer to something.  Not all operations in terraform are creating the resource, eg `terraform destroy -target module.bigquery_dataset`, is very weird with the prefix.

**Review by MXfive — CHANGES_REQUESTED** [review]

File: example/create_cloudsql_connection/main.tf (hunk 1 above)
Comment (JoseFMP) [reply example/create_cloudsql_connection/main.tf]: Thank you Martyn. I just renamed it.

**Review by MXfive — APPROVED** [review]

**Pull request merged**